src/TurnerSoftware.SitemapTools/SitemapQuery.cs

```diff
@@ -178,12 +179,12 @@ public async Task<SitemapFile> GetSitemapAsync(Uri sitemapUrl, CancellationToken
 								}
 							}
 						}
-						else
+						else if (!permitInvalidOperations)
 						{
 							throw new InvalidOperationException($"No sitemap readers for {sitemapType}");
 						}
 					}
-					else
+					else if (!permitInvalidOperations)
```

Comment:
I'm thinking more that we remove this entire `else` block - instead we treat sitemaps that we don't understand entirely as they don't exist. The code looks like it will fall through to a `return null` by removing the this `else` block.

Reply:
I'd had the same consideration but wasn't sure if you'd prefer to throw an exception vs returning null, which was why I gave the switch so the user can decide the behaviour.

Reply:
But to your point, if we can't understand it, then treating it as non-existent makes sense to me.